typebinder/src/exporters/check.rs

```diff
@@ -0,0 +1,133 @@
+use crate::exporters::utils::{get_file_contents, get_output_file_path};
+use crate::{
+    error::TsExportError,
+    exporters::{Exporter, HeaderComment},
+    pipeline::module_step::ModuleStepResultData,
+};
+use displaythis::Display;
+use std::collections::HashMap;
+use std::io::Read;
+use std::path::PathBuf;
+
+pub struct CheckExport {
+    root_path: PathBuf,
+    default_module_name: Option<String>,
+    header_comment: HeaderComment,
+    patches: HashMap<PathBuf, DiffPatch>,
+}
+
+impl Default for CheckExport {
+    fn default() -> Self {
+        let root_path = std::env::current_dir().expect("Failed to find current dir");
```

Comment:
I doubt it's a good idea to build default based on the context.. it can be a little bit misleading, because `default()` function is not pure in this case.
Also, the only place where it's used is in `CheckExport::new()`, where `root_path` gets overwritten.
I'd suggest remove this `Default` implementation and adjusting `CheckExport::new()` a little bit.